test/core/security/tls_security_connector_test.cc

```diff
@@ -376,6 +376,52 @@ TEST_F(TlsSecurityConnectorTest, TlsCheckHostNameFail) {
   GRPC_ERROR_UNREF(error);
 }
 
+TEST_F(TlsSecurityConnectorTest, CompareChannelSecurityConnectorSucceeds) {
+  grpc_core::RefCountedPtr<grpc_tls_certificate_distributor> distributor =
+      grpc_core::MakeRefCounted<grpc_tls_certificate_distributor>();
+  distributor->SetKeyMaterials(kRootCertName, root_cert_0_, absl::nullopt);
+  grpc_core::RefCountedPtr<::grpc_tls_certificate_provider> provider =
+      grpc_core::MakeRefCounted<TlsTestCertificateProvider>(distributor);
+  auto options = grpc_core::MakeRefCounted<grpc_tls_credentials_options>();
+  options->set_certificate_provider(provider);
+  options->set_watch_root_cert(true);
+  options->set_root_cert_name(kRootCertName);
+  grpc_core::RefCountedPtr<TlsCredentials> credential =
+      grpc_core::MakeRefCounted<TlsCredentials>(options);
+  grpc_core::RefCountedPtr<grpc_channel_security_connector> connector =
+      credential->create_security_connector(nullptr, kTargetName, nullptr,
+                                            nullptr);
+  EXPECT_EQ(connector->cmp(connector.get()), 0);
```

Comment:
This should actually create two different security connectors from the same credentials and then compare them against each other, rather than comparing a single SC against itself.

Same thing for the server SC test below.